From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Azure Dreams <azuredreams@users.noreply.github.com>
Date: Fri, 9 Oct 2026 15:18:01 +0000
Subject: [PATCH 2/5] Build with newer NDKs: include <functional> where
 std::function is used

libc++ in NDK 27 and later no longer includes it through other headers.
---
 libretrodroid/src/main/cpp/libretrodroid.h   | 1 +
 libretrodroid/src/main/cpp/rumble.h          | 1 +
 libretrodroid/src/main/cpp/utils/javautils.h | 1 +
 3 files changed, 3 insertions(+)

diff --git a/libretrodroid/src/main/cpp/libretrodroid.h b/libretrodroid/src/main/cpp/libretrodroid.h
index b507aee..03bf835 100644
--- a/libretrodroid/src/main/cpp/libretrodroid.h
+++ b/libretrodroid/src/main/cpp/libretrodroid.h
@@ -18,6 +18,7 @@
 #ifndef LIBRETRODROID_LIBRETRODROID_H
 #define LIBRETRODROID_LIBRETRODROID_H
 
+#include <functional>
 #include <jni.h>
 
 #include <EGL/egl.h>
diff --git a/libretrodroid/src/main/cpp/rumble.h b/libretrodroid/src/main/cpp/rumble.h
index 354ec2f..6f5751e 100644
--- a/libretrodroid/src/main/cpp/rumble.h
+++ b/libretrodroid/src/main/cpp/rumble.h
@@ -18,6 +18,7 @@
 #ifndef LIBRETRODROID_RUMBLE_H
 #define LIBRETRODROID_RUMBLE_H
 
+#include <functional>
 #include <array>
 
 #include "rumblestate.h"
diff --git a/libretrodroid/src/main/cpp/utils/javautils.h b/libretrodroid/src/main/cpp/utils/javautils.h
index 3ff492a..91b0bb9 100644
--- a/libretrodroid/src/main/cpp/utils/javautils.h
+++ b/libretrodroid/src/main/cpp/utils/javautils.h
@@ -18,6 +18,7 @@
 #ifndef LIBRETRODROID_JAVAUTILS_H
 #define LIBRETRODROID_JAVAUTILS_H
 
+#include <functional>
 #include "unordered_map"
 #include "string"
 
